.github/workflows/package.yaml

```diff
@@ -20,24 +20,31 @@ jobs:
         with:
           pixi-version: v0.41.4
           manifest-path: pyproject.toml
-      - name: build pypi package
+      - name: Verify Mantid version
         run: |
-          pixi run build-pypi
-      - name: build conda package
+          pixi run verify-mantid-version-dev
+      - name: Test production environment
+        run: |
+          pixi install --environment prod
+          pixi run --environment prod test-ci
+          pixi run --environment prod verify-mantid-version
+      - name: Test QA environment
+        run: |
+          pixi install --environment qa
+          pixi run --environment qa test-ci
+          pixi run --environment qa verify-mantid-version-qa
+      - name: Build conda package
         run: |
           pixi run build-conda
-      - name: upload conda package to anaconda
+      - name: Upload conda package (on tags)
+        # Upload to neutrons conda channel on anaconda.org
         if: startsWith(github.ref, 'refs/tags/v')
         env:
           ANACONDA_API_TOKEN: ${{ secrets.ANACONDA_TOKEN }}
-          IS_RC: ${{ contains(github.ref, 'rc') }}
         run: |
-          # label is main or rc depending on the tag-name
-          CONDA_LABEL="main"
-          if [ "${IS_RC}" = "true" ]; then CONDA_LABEL="rc"; fi
-          echo pushing ${{ github.ref }} with label $CONDA_LABEL
-          pixi run anaconda upload --label $CONDA_LABEL --user neutrons snapred-*.conda
+          pixi run publish-conda
```

Comment:
can you just revert the code?  I dont see how the new code takes into account which channel to publish to.